### PR TITLE
Move webhook factories to core/lib

### PR DESCRIPTION
**Pull request by aleksandarpetrushev**

### CODE REVIEW

File: api/lib/spree/api.rb [hunk 1]
```diff
@@ -10,3 +10,7 @@ module Api
 
 require 'spree/api/api_dependencies'
 require 'spree/api/engine'
+
+Dir["#{File.dirname(__FILE__)}/factories/**"].each do |f|
```
Comment (damianlegawiec) [inline api/lib/spree/api.rb:14]: @aleksandarpetrushev why code duplication? you can just require `testing_support/factories.rb` here

Comment (aleksandarpetrushev) [reply api/lib/spree/api.rb:14]: I seem to get NameError: uninitialized constant Spree::Webhooks

Comment (damianlegawiec) [reply api/lib/spree/api.rb:14]: create a `api/lib/spree/webhooks.rb` and define the module there, and also you can move the requirement there

Comment (aleksandarpetrushev) [reply api/lib/spree/api.rb:14]: I would also need to define ::Events and ::Subsciber modules as well. And any others added in the future will need to be defined

Comment (aleksandarpetrushev) [reply api/lib/spree/api.rb:14]: Actually looks like we can not do it that way as well

**Comment by aleksandarpetrushev** [comment]
@damianlegawiec I will open a new PR for this, because there are conflicts, when I merged with main there are too many commits and it will become messy

**Pull request closed without merging**